ARCHITECTURE.md

```diff
@@ -0,0 +1,3386 @@
+
+        let hours_passed = total_minutes / 60;
+        let total_hours = self.time.hour as u32 + hours_passed;
+        self.time.hour = (total_hours % 24) as u8;
+
+        let days_passed = total_hours / 24;
+        self.time.day += days_passed;
+
+        // Update season every 90 days
+        let season_day = self.time.day % 360;
+        self.time.season = match season_day {
+            0..=89 => Season::Spring,
+            90..=179 => Season::Summer,
+            180..=269 => Season::Autumn,
+            _ => Season::Winter,
+        };
+    }
+}
+```
+
+---
+
+### 2.2 Crate #2: `narrative_core` (The Cortex)
+
+This crate is the "brain" of the system - it interfaces with `game_rules`, communicates with LLMs, and manages narrative generation.
+
+#### 2.2.1 Module: `knowledge_base`
+
+**Purpose:** Long-term memory storage using an associative knowledge graph.
+
+```mermaid
+graph TD
+    subgraph "Knowledge Graph Structure"
+        T1[Tag: Villain_A]
+        T2[Tag: Priestess_B]
+        T3[Tag: Orphanage]
+        T4[Tag: Magic]
+        T5[Tag: Combat]
+
+        F1[Fact: Grew up together]
+        F2[Fact: Both have magical abilities]
+        F3[Fact: Villain betrayed the order]
+
+        T1 -->|0.9| T3
+        T2 -->|0.9| T3
+        T1 -->|0.7| T4
+        T2 -->|0.8| T4
+        T1 -->|0.6| T5
+
+        F1 -.->|tagged| T1
+        F1 -.->|tagged| T2
+        F1 -.->|tagged| T3
+
+        F2 -.->|tagged| T1
+        F2 -.->|tagged| T2
+        F2 -.->|tagged| T4
+
+        F3 -.->|tagged| T1
+    end
+```
+
+**Tag and Fact Definitions:**
+
+```rust
+// src/knowledge_base/types.rs
+
+use serde::{Deserialize, Serialize};
+use std::collections::{HashMap, HashSet};
+use uuid::Uuid;
+
+/// Tags are the nodes in our knowledge graph
+/// They represent concepts, entities, locations, or themes
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum Tag {
+    /// Reference to a specific entity
+    Entity(EntityId),
+
+    /// A location in the world
+    Location(LocationId),
+
+    /// A concept or theme
+    Concept(String),
+
+    /// A faction or organization
+    Faction(String),
+
+    /// An event type
+    EventType(String),
+
+    /// A relationship type
+    RelationType(String),
+
+    /// Custom tag for extension
+    Custom(String),
+}
+
+impl Tag {
+    pub fn as_string(&self) -> String {
+        match self {
+            Tag::Entity(id) => format!("entity:{}", id.0),
+            Tag::Location(id) => format!("location:{}", id.0),
+            Tag::Concept(s) => format!("concept:{}", s),
+            Tag::Faction(s) => format!("faction:{}", s),
+            Tag::EventType(s) => format!("event:{}", s),
+            Tag::RelationType(s) => format!("relation:{}", s),
+            Tag::Custom(s) => format!("custom:{}", s),
+        }
+    }
+}
+
+/// Unique identifier for facts
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct FactId(pub Uuid);
```

Comment:
why should we store this as UUID, just numeric id is ussually better to store in structs like `IndexVec`